core/src/main/java/com/linecorp/armeria/client/retry/RetryConfigBuilder.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.retry;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+
+import javax.annotation.Nullable;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.MoreObjects.ToStringHelper;
+
+import com.linecorp.armeria.common.Flags;
+import com.linecorp.armeria.common.Response;
+
+/**
+ * Builds a {@link RetryConfig}.
+ * A {@link RetryConfig} instance encapsulates the used {@link RetryRule}, maxTotalAttempts,
+ * and responseTimeoutMillisForEachAttempt.
+ */
+public final class RetryConfigBuilder<T extends Response> {
+    private int maxTotalAttempts = Flags.defaultMaxTotalAttempts();
+    private long responseTimeoutMillisForEachAttempt = Flags.defaultResponseTimeoutMillis();
+    private int maxContentLength;
+
+    @Nullable
+    private final RetryRule retryRule;
+
+    @Nullable
+    private final RetryRuleWithContent<T> retryRuleWithContent;
+
+    /**
+     * Returns a {@link RetryConfigBuilder} with this {@link RetryRule}.
+     */
+    RetryConfigBuilder(RetryRule retryRule) {
+        this.retryRule = requireNonNull(retryRule, "retryRule");
+        retryRuleWithContent = null;
+        maxContentLength = 0;
+    }
+
+    /**
+     * Returns a {@link RetryConfigBuilder} with this {@link RetryRuleWithContent}.
+     */
+    RetryConfigBuilder(RetryRuleWithContent<T> retryRuleWithContent) {
+        retryRule = null;
+        this.retryRuleWithContent = requireNonNull(retryRuleWithContent, "retryRuleWithContent");
+        maxContentLength = Integer.MAX_VALUE;
+    }
+
+    /**
+     * Sets the maxContentLength to be used with a {@link RetryRuleWithContent}.
+     */
+    public RetryConfigBuilder<T> maxContentLength(int maxContentLength) {
+        checkArgument(maxContentLength > 0,
```

Comment:
How about raising an exception if `retryRule != null`? or
We can add this parameter to the constructor that takes `RetryRuleWithContent`.

Reply:
I had it in the constructor at the beginning and moved it to a setter based on an earlier comment.
Added requireNonNull for retryRuleWithContent.

Reply:
Ah, I missed that. 😅